src/v/cluster/cluster_discovery.cc

```diff
@@ -131,8 +136,39 @@ cluster_discovery::dispatch_node_uuid_registration_to_seeds(
     co_return ss::stop_iteration::no;
 }
 
-bool cluster_discovery::is_cluster_founder() const {
-    return config::node().seed_servers().empty();
+/*static*/ std::optional<node_id>
+cluster_discovery::get_cluster_founder_node_id() {
+    if (config::node().empty_seed_starts_cluster()) {
+        if (config::node().seed_servers().empty())
+            return node_id{0};
+        else
+            return {};
+    } else {
+        if (auto idx = get_node_index_in_seed_servers(); idx)
+            return node_id{*idx};
+        else
+            return {};
+    }
+}
+
+/*static*/ std::optional<int32_t>
+cluster_discovery::get_node_index_in_seed_servers() {
+    const std::vector<config::seed_server>& seed_servers
+      = config::node().seed_servers();
+    vassert(
+      !seed_servers.empty(),
+      "Configuration error: seed_servers cannot be empty when "
+      "empty_seed_starts_cluster is false");
+    const auto it = std::find_if(
+      seed_servers.cbegin(),
+      seed_servers.cend(),
+      [rpc_addr = config::node().advertised_rpc_api()](
+        const config::seed_server& seed_server) {
+          return rpc_addr == seed_server.addr;
+      });
+    if (it == seed_servers.cend()) return {};
```

Comment:
curly braces please

Reply:
np, I will check though all my code for that